### PR TITLE
Endian-ness enforcement of generated compressed fabric ID

### PR DESCRIPTION
**Pull request by pan-apple**
#### Problem
The generated compressed fabric ID may be different for the same input for different platforms based on endianness.

#### Change overview
Convert generated compressed fabric ID to host system.

#### Testing
Added a new unit test that checks for generated compressed fabric ID against expected values for a given set of test inputs.

### CODE REVIEW

File: src/transport/FabricTable.cpp [hunk 1]
```diff
@@ -174,9 +174,13 @@ CHIP_ERROR FabricInfo::FetchFromKVS(PersistentStorageDelegate * kvs)
 CHIP_ERROR FabricInfo::GetCompressedId(FabricId fabricId, NodeId nodeId, PeerId * compressedPeerId) const
 {
     ReturnErrorCodeIf(compressedPeerId == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    CompressedFabricId compressedFabricId = 0;
-    MutableByteSpan fabricIdSpan(reinterpret_cast<uint8_t *>(&compressedFabricId), sizeof(compressedFabricId));
-    ReturnErrorOnFailure(GenerateCompressedFabricId(mRootPubkey, fabricId, fabricIdSpan));
+    uint8_t compressedFabricIdBuf[sizeof(CompressedFabricId)];
```
Comment (andy31415) [inline src/transport/FabricTable.cpp:177]: should we use sizeof(uint64_t) here to make it explicit?

Using CompressedFabricId makes it look a lot like a strong and below I see that it is just a Get64.

**Review by andy31415 — APPROVED** [review]

File: src/transport/tests/TestFabricTable.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+};
+
+void TestGetCompressedFabricID(nlTestSuite * inSuite, void * inContext)
+{
+    FabricInfo fabricInfo;
+
+    fabricInfo.SetRootCert(ByteSpan(sTestRootCert));
+
+    PeerId compressedId;
+    NL_TEST_ASSERT(inSuite, fabricInfo.GetCompressedId(1234, 4321, &compressedId) == CHIP_NO_ERROR);
```
Comment (tcarmelveilleux) [inline src/transport/tests/TestFabricTable.cpp:60]: I just realized looking at this test that we are loading the root cert in memory in the fabric table **and keeping it there**. Please open an issue for SetRootCert not to hold on to the actual root cert (for which only the public key and SKID are needed to be kept, which amount to 85 bytes total, if you keep the public key cached, and 20 bytes if you only keep the SKID).

**Review by jmartinez-silabs — APPROVED** [review]

File: src/transport/tests/TestFabricTable.cpp (hunk 2 above)
Comment (pan-apple) [reply src/transport/tests/TestFabricTable.cpp:60]: I have an overarching issue for all the storage optimizations in FabricInfo and table.
https://github.com/project-chip/connectedhomeip/issues/9131

Comment (pan-apple) [reply src/transport/tests/TestFabricTable.cpp:60]: I have added a TODO specific to Root Cert and NOC, just so that it doesn't get overlooked.

**Review by woody-apple — APPROVED** [review]

**Pull request merged**